### ISSUE TITLE
Secure my api from openid token

### ISSUE THREAD

**Issue by jitendersaini** [djpugh/fastapi_aad_auth#64]
I didn't get much from the doc to secure an API with azure openid token. Lets say my web/mobile client is sending Bearer token in header and I want my api to be secured e.g.

app = FastAPI()

app.config['AZURE_OAUTH_TENANCY'] = 'xxx'
app.config['AZURE_OAUTH_APPLICATION_ID'] = 'xxx'

auth = Authenticator()
auth.init_app(app)

@app.route('/unprotected')
def unprotected():
    return 'hello world'

@app.route('/protected')
@auth()
def protected():
    return 'hello authenticated entity'

@app.route('/protected-with-single-scope')
@auth('required-scope')
def protected_with_scope():
    return 'hello authenticated and authorised entity'

@app.route('/protected-with-multiple-scopes')
@auth('required-scope1 required-scope2')
def protected_with_multiple_scopes():
    return 'hello authenticated and authorised entity'

I'm looking for some declarative like @auth or some kind of configuration which will secure my /helllo api at backend.

**Comment by djpugh** [comment]
Hi @jitendersaini, thanks for the question and a good idea to add a decorator syntax for fastapi protection and scope verification - I'll look to add it this weekend.

You can use the [fastapi dependency injection approach](https://fastapi.tiangolo.com/tutorial/dependencies/)

```
@router.get('/hello')
async def hello_world(auth_state: AuthenticationState = Depends(auth_provider.auth_backend.requires_auth(allow_session=True))):
    print(auth_state)
    return {'hello': 'world'}
```

From the [docs](https://djpugh.github.io/fastapi_aad_auth/usage.html#including-fastapi-aad-auth-in-your-code).

If you need to check scopes, we currently aren't validating them in the token auth directly, but they are provided in the User object in the authentication state, which you can check, and I will also add some clarifications to the docs on the options here.

**Comment by djpugh** [comment]
I don't think I've tested this fully, but

https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/

https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-in-path-operation-decorators/

May work and will be what I'll plan to integrate with.

**Comment by jitendersaini** [comment]
Thanks for your prompt response djpugh, I'll check and get back here.

**Comment by jitendersaini** [comment]
Hello again, It will really be glad to see if it could be quite simple like one declarative action like @auth() on any method for securing it. Actually my app is getting token from some azure login URL and sending that token to my backend system where I want to validate the token before accessing the required method. Just configuring my backend with application id and tenant id the passed Bearer token should be recognised by @auth decorator like it's happening in https://pypi.org/project/flask-azure-oauth/. 
Thanks

**Comment by djpugh** [comment]
Apologies it took a while, but the fix in #70 should sort this
